packages/components/src/reaction_selection_menu/index.jsx

```diff
@@ -0,0 +1,96 @@
+import React, { useCallback, useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import clsx from 'clsx';
+import { invoke } from 'nebenan-helpers/lib/utils';
+import styles from './index.module.scss';
+import ReactionIcon from '../reaction_icon';
+import { useActiveEventListener, useLongHover, useLongTouch } from './hooks';
+import { getItemIndexForPositionFromElements } from './utils';
+import { REACTIONS } from '../constants';
+
+const ReactionSelectionMenu = ({ className, label, strings, onSelect }) => {
+  const rootRef = useRef(null);
+  const [isActive, setIsActive] = useState(false);
+  const [hoverReaction, setHoverReaction] = useState(null);
+  const itemRefs = useRef([]);
+
+  const { active: activeTouchEvents, passive: touchEvents } = useLongTouch({
+    onStart: useCallback(() => {
+      setIsActive(true);
+    }, []),
+    onEnd: useCallback(() => {
+      setHoverReaction(null);
+      setIsActive(false);
+
+      if (hoverReaction) invoke(onSelect, hoverReaction);
+    }, [hoverReaction, onSelect]),
+    onMove: useCallback((event) => {
+      const { clientX, clientY } = event.touches[0];
```

Comment:
this is iffy across devices, you might want to use `eventCoordinates` from `nebenan-helpers/lib/dom`